### ISSUE TITLE
Refactor the flow definition in the i/o and the model

### ISSUE THREAD

**Issue by datejada** [TulipaEnergy/TulipaEnergyModel.jl#183]
### Summary

The function `create_model` defines the flows from the graph inside:

```
 F   = [(A[e.src], A[e.dst]) for e ∈ edges(graph)] # f[1] -> source, f[2] -> destination
 Fi  = [f for f ∈ F if params.flows_investable[f]]
 Ft  = [f for f ∈ F if params.flows_is_transport[f]]

```

This makes it not standard with assets definition, for instance:

```
 A   = sets.assets
 Ac  = sets.assets_consumer
```

### Basic Example

We can refactor the code so they look like the same as the:

```
 F   = sets.flows
 Fi  = sets.flows_investment
 Ft  = sets.flows_transport

```

### Drawbacks

Need to move the graph into the i/o and refactor the tests

**Comment by clizbe** [comment]
I'm trying to implement this, but realize I don't understand our model structure. Good practice I guess.
Also, certain terms are starting to run into each other. For example:
flows_investment (param)
flows_investment (set)
flows_investable (var)

Not sure what's the plan to handle this?

**Comment by clizbe** [comment]
Hmm I also ran into the problem that (I think) the graph needs to be created before creating the flows set. So I think this is just beyond my understanding. Maybe good to walk through the model at some point so I have a better idea.

**Comment by datejada** [comment]
@clizbe, I agree that we need to tackle the naming convention or define a good practice somewhere to avoid confusion in the names (especially in the future when we don't remember why the name :sweat_smile:)

Regarding the second question, it is the drawback in the issue that we need to move the graph function before the creation of parameters and assets. Moreover, pass the graph as an argument to the function that creates the parameters and sets.

So, in the test files:

```
graph = create_graph(joinpath(dir, "assets-data.csv"), joinpath(dir, "flows-data.csv"))
parameters, sets = create_parameters_and_sets_from_file(dir, graph)
```
 Look what the differences are, the order and a new argument in the `create_parameters_and_sets_from_file`

Then you need to add the argument in the `create_parameters_and_sets_from_file` function, something like this:

```
function create_parameters_and_sets_from_file(input_folder::AbstractString, graph)
```

So, now you can call the graph structure in the parameters.

In addition, we won't use the graph argument in the `create_model` function so that you can take it from there too.

Finally, remember to update the function descriptions (the text above each function)

Does this make sense? Do you think you can try it again? I have faith in you 😄

**Comment by abelsiqueira** [comment]
Notice that the names must always be prefixed by the variable that holds them, so it's `params.x` and `sets.x`. That being said, I would prefer a more descriptive name as well.

On the graph creation thing, it makes sense to create the graph in the same function as the rest of the things, because the files being read are the same. So maybe a single function

```julia
parameters, sets, graph = create_structures_from_data_folder(input_folder)
```

**Comment by datejada** [comment]
waiting on https://github.com/TulipaEnergy/TulipaEnergyModel.jl/issues/110

**Comment by datejada** [comment]
Solved in PR #254
